app/src/main/java/com/crossbowffs/nekosms/loader/InboxSmsLoader.java

```diff
@@ -33,7 +33,7 @@ private static ContentValues serializeMessage(SmsMessageData messageData) {
     }
 
     public static Uri writeMessage(Context context, SmsMessageData messageData) {
-        if (!AppOpsUtils.noteOp(context, AppOpsUtils.OP_WRITE_SMS)) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R && !AppOpsUtils.noteOp(context, AppOpsUtils.OP_WRITE_SMS)) {
```

Comment:
If the signature is wrong on Android R, this version check should be moved inside `noteOp`. Or even better, just delete it altogether if it doesn't do anything.